sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -239,6 +239,27 @@ def resolved_hostname(self):
             elif self.dbm_enabled:
                 host, port = self.split_sqlserver_host_port(self.instance.get('host'))
                 self._resolved_hostname = resolve_db_host(host)
+                engine_edition = self.static_info_cache.get(STATIC_INFO_ENGINE_EDITION)
+                if engine_edition == ENGINE_EDITION_SQL_DATABASE:
+                    configured_database = self.instance.get('database', None)
+                    if not configured_database:
+                        configured_database = 'master'
+                        self.warning(
+                            "Missing 'database' in instance configuration."
+                            "For Azure SQL Database a non-master application database must be specified."
+                        )
+                    elif configured_database == 'master':
+                        self.warning(
+                            "Wrong 'database' configured."
+                            "For Azure SQL Database a non-master application database must be specified."
+                        )
+                    azure_server_suffix = ".database.windows.net"
+                    if host.endswith(azure_server_suffix):
+                        host = host[: -len(azure_server_suffix)]
```

Comment:
Why remove the suffix?

Reply:
To match what Azure shows in the portal. 

Example:
<img width="750" alt="image" src="https://user-images.githubusercontent.com/3707657/175363783-48eb6aeb-76d0-4ea5-a454-5482f873f588.png">

Server names are globally unique across subscriptions so the suffix doesn't add any information.